### PR TITLE
get mock impl from oas examples

### PR DESCRIPTION
**Pull request by AmaliMatharaarachchi**
### Purpose
Adding mock impl endpoints using oas examples.
Examples:
```
/testMockApi:
    get:
      summary: Tests Choreo Connect mock API implementation
      operationId: testImpl
      responses:
        '200':
          description: Expected response to a valid request
          content:
            text/html:
              examples:
                example1:
                  value:
                    description: content for example 1
                example2:
                  value:
                    description: content for example 2
          headers:
            x-wso2-header:
              schema:
                type: string
              example: Sample header value
        'default':
          description: Expected response to a valid request
          content:
            application/json:
              example:
                description: default content
          headers:
            x-wso2-header:
              schema:
                type: string
              example: Sample header value
```
specify the example with "Prefer" header
Example:
`-H "Prefer: code=200, example=example1"`

### Issues

Fixes https://github.com/wso2/product-microgateway/issues/2686

### Automation tests
 - Unit tests added: No
 - Integration tests added: Yes

### Tested environments

Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/internal/oasparser/model/api_yaml.go [hunk 1]
```diff
@@ -134,15 +134,17 @@ func NewAPIYaml(fileContent []byte) (apiYaml APIYaml, err error) {
...
+	}
 	err = apiYaml.ValidateMandatoryFields()
 	if err != nil {
 		loggers.LoggerAPI.Errorf("%v", err)
 		return apiYaml, err
 	}
 
 	if apiYaml.Data.EndpointImplementationType == constants.InlineEndpointType {
-		errmsg := "inline endpointImplementationType is not currently supported with Choreo Connect"
+		errmsg := "inline endpointImplementationType is not supported with Choreo Connect"
```
Comment (slahirucd7) [inline adapter/internal/oasparser/model/api_yaml.go:147]: Can we change `inline ` -> `Inline`

File: adapter/internal/oasparser/model/api_operation.go [hunk 2]
```diff
@@ -37,34 +44,135 @@ type Operation struct {
...
+		}
+	}
 }
 
-// MockedContentConfig holds mocked content configurations in the mocked API operation
-type MockedContentConfig struct {
-	ContentType string `json:"contentType,omitempty"`
-	Body        string `json:"body,omitempty"`
+// asJSON parse interface to son string. returns error if a null value has passed
+func asJSON(data interface{}) (string, error) {
```
Comment (slahirucd7) [inline adapter/internal/oasparser/model/api_operation.go:167]: Can we change this method name to `convertJson` ? There's a small typo in the comment.

File: adapter/internal/oasparser/model/mgw_swagger.go [hunk 3]
```diff
@@ -638,41 +639,43 @@ func (swagger *MgwSwagger) setDisableSecurity() {
...
-	if err != nil {
-		logger.LoggerOasparser.Errorf("Error while parsing the sandbox endpoints of the API %s:%s - %v",
-			swagger.title, swagger.version, err)
-		return err
-	}
-	for _, res := range swagger.resources {
-		err := res.productionEndpoints.validateEndpointCluster("Resource level production")
+			logger.LoggerOasparser.Errorf("No Endpoints are provided for the API %s:%s",
+				swagger.title, swagger.version)
+			return errors.New("no endpoints are provided for the API")
```
Comment (slahirucd7) [inline adapter/internal/oasparser/model/mgw_swagger.go:648]: `No endpoints.` can we change this as well ?

Comment (AmaliMatharaarachchi) [reply adapter/internal/oasparser/model/mgw_swagger.go:648]: golint asks to create errors objects starting with lowercase letters

**Review by pubudu538 — APPROVED** [review]

**Review by slahirucd7 — APPROVED** [review]

**Pull request merged**